### ISSUE TITLE
Auto-generate Neovim 0.5.0 API 7

### ISSUE THREAD

**Issue by jgehrig** [equalsraf/neovim-qt#698]
After Neovim's 0.5.0 release, we should generate the new API. We should wait until the API is finalized.

Pull Request #697 added support for Float values. We should validate this new code. Add unit tests?

**Comment by equalsraf** [comment]
I made a small change recently to allow the binding generator to read mpack files. So we don't need to run whatever neovim version introduced the api v7. The neovim repo hosts the .mpack files for the different versions.

**Comment by equalsraf** [comment]
Need to figure out the new LuaRef type being used in API7. Unlike the other ext types I don't see it being typedef explicitly.

```
Found unsupported type(LuaRef) when adding function nvim_win_call(), skipping
Found unsupported type(LuaRef) when adding function nvim_buf_call(), skipping

```

**Comment by equalsraf** [comment]
I was confused for a bit but it seems like the LuaRef type was originally encoded as a NIL msgpack value. The value itself is not explicitly used in output of any neovim API function but it does appear in its dictionary type e.g. this returns one in its "callback" attribute

```
ArrayOf(Dictionary) nvim_get_keymap(String mode, )
```

The encoding changed upstream when neovim/neovim#41065 was merged and now it is encoded as a string. This much I can verify by looking a the callback attribute for keymap i.e. if I set 

```
vim.api.nvim_set_keymap('n', '<leader>b', '', {noremap = true, desc = 'Test', callback = function() end})
```

and then call nvim_get_keymap, the value of **callback** will be:

- `QVariant(Invalid))` for nvim 0.12.1 this is the msgpac NIL
- `QVariant(QByteArray`, "<Lua 4>") for the latest upstream, this is a string

So for now I think I will just skip generation of api functions that use LuaRef because

1. only two functions use it as input (nvim_win_call, nvim_buf_call) and we have no use for them ATM
2. this seems like an upcoming breaking change upstream, but NIL would not be useful when passed to those functions and I don't think the upstream PR changes that
3. there is no api to explicitly create a LuaRef
4. the docs mention that these two apis are `Lua |vim.api| only`

cc @justinmk

**Comment by justinmk** [comment]
> only two functions use it as input (nvim_win_call, nvim_buf_call) and we have no use for them ATM

yeah... there are some functions like `nvim_create_autocmd(..., {callback=...})` which also accept luarefs but in a less "formal" way?



> 4\. these two apis are `Lua |vim.api| only`

it may become more common; one idea i've been mulling over, is: API called via Lua can pass a lua function, else it accepts a Excmd string (i.e. this would be a standard pattern with maybe a standard RPC "type").
